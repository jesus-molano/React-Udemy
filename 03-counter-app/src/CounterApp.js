import { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 0 }) => {

  const [counter, setCounter] = useState(value)

  // handles
  const handleAdd = () => setCounter(counter + 1);
  const handleSub = () => setCounter(counter - 1);
  const handleRes = () => setCounter(value);
  // Otro manejo de setState
  // const handleSub = () => setCounter((state) => state - 1); //(valor anterior) => nuevo valor

  return (
    <>
      <h1>CounterApp</h1>
      <p>{counter}</p>
      <button onClick={handleAdd} >+</button>
      <button onClick={handleRes} >Reset</button>
      <button onClick={handleSub} >-</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number,
}

export default CounterApp;