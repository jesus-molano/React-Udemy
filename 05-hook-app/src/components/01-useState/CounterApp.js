import React, { useState } from 'react';
import './counter.css';

export const CounterApp = () => {
  const [counters, setCounters] = useState({ counter1: 10, counter2: 20 });
  const { counter1, counter2 } = counters;
	return (
		<>
			<h1>Counter 1: {counter1}</h1>
			<h1>Counter 2: {counter2}</h1>
			<hr />
			<button
				className='btn btn-primary'
				onClick={() => {
          setCounters({
            ...counters,
            counter1: counter1 + 1,
          });
				}}
			>
				+
			</button>
		</>
	);
};
