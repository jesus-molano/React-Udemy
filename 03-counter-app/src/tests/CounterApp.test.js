import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';

let wrapper = shallow(<CounterApp />);

beforeEach(() => {
  wrapper = shallow(<CounterApp />);
})

describe('CounterApp component tests', () => {
  test('should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
  test('should show the number 100 by default', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    const valorEsperado = wrapper.find('p').text()

    expect(valorEsperado).toBe(value.toString());

  })
  test('should increase with the button +', () => {
    wrapper.find('button').at(0).simulate('click');
    const valorEsperado = wrapper.find('p').text();

    expect(valorEsperado).toBe('1');
  })
  test('should show the default value with the button Reset', () => {
    const value = 100;
    const wrapper = shallow(<CounterApp value={value} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    const valorEsperado = wrapper.find('p').text();

    expect(valorEsperado).toBe(`${value}`);
  })
  test('should decrease with the button -', () => {
    wrapper.find('button').at(2).simulate('click');
    const valorEsperado = wrapper.find('p').text();

    expect(valorEsperado).toBe('-1');
  })
})
