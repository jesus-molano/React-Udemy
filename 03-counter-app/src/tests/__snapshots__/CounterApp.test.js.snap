// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp component tests should show the component correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
</Fragment>
`;
